internal/core/operations/remove_rooms/remove_rooms_executor.go

```diff
@@ -73,18 +91,67 @@ func (e *RemoveRoomsExecutor) Execute(ctx context.Context, op *operation.Operati
 			if errors.Is(err, serviceerrors.ErrGameRoomStatusWaitingTimeout) {
 				return operations.NewErrTerminatingPingTimeout(deleteErr)
 			}
-			return operations.NewErrUnexpected(deleteErr)
+
+			return operations.NewErrUnexpected(fmt.Errorf("failed to remove room by amount: %w", err))
 		}
 	}
 
 	logger.Info("finished deleting rooms")
 	return nil
 }
 
+func (e *RemoveRoomsExecutor) removeRoomsByIDs(ctx context.Context, schedulerName string, roomsIDs []string) error {
+	rooms := make([]*game_room.GameRoom, 0, len(roomsIDs))
+	for _, roomID := range roomsIDs {
+		gameRoom, err := e.roomStorage.GetRoom(ctx, schedulerName, roomID)
+		if err != nil {
+			return err
+		}
+
+		rooms = append(rooms, gameRoom)
+
+	}
+
+	err := e.deleteRooms(ctx, rooms)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *RemoveRoomsExecutor) removeRoomsByAmount(ctx context.Context, schedulerName string, amount int) error {
+	rooms, err := e.roomManager.ListRoomsWithDeletionPriority(ctx, schedulerName, "", amount, &sync.Map{})
+	if err != nil {
+		return err
+	}
+
+	err = e.deleteRooms(ctx, rooms)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *RemoveRoomsExecutor) deleteRooms(ctx context.Context, rooms []*game_room.GameRoom) error {
+	var err error
+	for _, room := range rooms {
```

Comment:
@arthur29 do you think it runs too much out of scope to make this deletion concurrent (using goroutines to delete rooms)?